staging/kos/pkg/registry/network/subnet/strategy.go

```diff
@@ -139,15 +137,17 @@ func (ss *subnetStrategy) checkNSAndCIDRConflicts(candidate *subnet.Summary, mal
 	glog.V(5).Infof("Found %d subnets with vni %d", len(potentialRivals), candidate.VNI)
 	// Check whether there are Namespace and CIDR conflicts with other subnets.
 	for _, potentialRival := range potentialRivals {
-		// Ignore the error because a malformed subnet would not have been
-		// allowed: it's the code in this file that performs validation.
-		pr, _ := subnet.NewSummary(potentialRival)
-
+		pr, err := subnet.NewSummary(potentialRival)
+		if err != nil {
```

Comment:
How could `err != nil`?  Only subnets that pass validation will appear in `protentialRivals`, right?

Reply:
We already discussed this in a past PR comment. What you said is true under the assumption that there never was a more lenient version of the extension API server letting through invalid Subnets. You said not to worry about that because we're not in production --- I agree, but most of the changes in this PR had been already coded and stashed away as I was doing more important stuff --- this change here was one of such changes, since it was already there I kept it.

Reply:
Do you want me to remove the check?

Reply:
This is going to look unnecessary to future reviewers, so yes, I would prefer it to not be here.  This can be addressed in a later PR.